airbyte-integrations/connectors/destination-langchain/destination_langchain/embedder.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+from destination_langchain.config import EmbeddingConfigModel
+from langchain.embeddings.base import Embeddings
+from langchain.embeddings.openai import OpenAIEmbeddings
+
+class Embedder(ABC):
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def check(self) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def langchain_embeddings(self) -> Embeddings:
+        pass
+
+    @property
+    @abstractmethod
+    def embedding_dimensions(self) -> int:
+        pass
+
+class OpenAIEmbedder(Embedder):
+    def __init__(self, config: EmbeddingConfigModel):
+        super().__init__()
+        self.embeddings = OpenAIEmbeddings(openai_api_key=config.openai_key, chunk_size=8191)
+    
+    def check(self) -> Optional[str]:
+        try:
+            self.embeddings.embed_query("test")
+        except Exception as e:
+            return str(e)
+        return None
+    
+    @property
+    def langchain_embeddings(self) -> Embeddings:
+        return self.embeddings
+
+    @property
+    def embedding_dimensions(self) -> int:
+        return 1536
```

Comment:
why?

Reply:
That's the size of the embedding vectors of the openai model, added a comment to explain